contracts/cw721-fixed-price/src/contract.rs

```diff
@@ -160,23 +160,21 @@ pub fn execute_receive(
         extension: config.extension.clone(),
     });
 
-    let callback = CosmosMsg::Wasm(WasmMsg::Execute {
-        contract_addr: config.cw20_address.to_string(),
-        msg: to_binary(&mint_msg)?,
-        funds: vec![],
-    });
-
-    config.unused_token_id += 1;
-    CONFIG.save(deps.storage, &config)?;
+    if let Some(cw721) = config.cw721_address.clone() {
+        let callback = Cw721Contract(cw721).call(mint_msg)?;
+        config.unused_token_id += 1;
+        CONFIG.save(deps.storage, &config)?;
 
-    Ok(Response::new().add_message(callback))
+        return Ok(Response::new().add_message(callback));
+    }
+    Err(ContractError::Cw721NotLinked {})
```

Comment:
Looks good, but tiny nit - shouldn't this block more idiomatically just be a match?

Reply:
Yeah, that would be nicer.